lib/src/features/dashboard_page.dart

```diff
@@ -11,35 +14,96 @@ class DashboardPage extends StatefulWidget {
   State<DashboardPage> createState() => _DashboardPageState();
 }
 
-class _DashboardPageState extends State<DashboardPage> {
+class _DashboardPageState extends State<DashboardPage>
+    with SingleTickerProviderStateMixin {
+  late TabController tabController;
+
   @override
-  Widget build(BuildContext context) {
-    return DefaultTabController(
-        length: 4,
-        child: Scaffold(
-        appBar: AppBar(
-          title: Text("Whatsapp"),
-          bottom: TabBar(
-            tabs:[
-              Tab(icon: Image.asset('assets/images/community_icon.png', color: Colors.white, width: 30, height: 20,),),
-              Tab(text: "Chats",),
-              Tab(text: "Status",),
-              Tab(text: "Calls",),
-          ]
-          )
-        ),
-        body: SafeArea(
-            child: TabBarView(
-                children:[
-                  Community(),
-                  Chats(),
-                  Status(),
-                  Calls()
-                ]
+  void initState() {
+    super.initState();
+    tabController = TabController(length: 4, vsync: this, initialIndex: 0);
+  }
 
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+          backgroundColor: wpgreen,
+          title: Text(Strings.appTitle),
+          actions: [
+            IconButton(onPressed: () {}, icon: Icon(Icons.search)),
+            PopupMenuButton<MenuOptions>(
+              itemBuilder: (BuildContext context) => [
+                const PopupMenuItem(
```

Comment:
It will be better if you use an enum for making the PopupMenuItems and rendering UI with the enum values, it gives you a cleaner code and easy to update. You can use ChatGPT to see how it works